File: Data Structures & Algorithms/string-encode-and-decode/submission-4.cpp
class Solution {
public:

    string encode(vector<string>& strs) {
        string encoded_string = "";
        for(auto & cha:strs){
            
            encoded_string+=to_string(cha.size());
            encoded_string+="?";
            encoded_string+=cha;

        }
        cout<<encoded_string;
        return encoded_string;
    }

    vector<string> decode(string s) {
        vector<string> decoded_strs;
        int count = 0;
        int index = 0;
        while(index<s.size()){
            count = index;
            while(s[count]!='?'){
                count++;
            }
            
           
            int len = stoi(s.substr(index, count-index));
            decoded_strs.push_back(s.substr(count+1, len));
           index = len+count+1;}
        
        return decoded_strs;
    }
};

File: Data Structures & Algorithms/remove-duplicates-from-sorted-array/submission-4.py
class Solution:
    def removeDuplicates(self, nums: List[int]) -> int:
        l, r = 1, 1
        counter = 0
        for i in nums:
            if nums.count(i) == 1:
                counter+=1
        while r < len(nums):
            if nums[r] != nums[r-1]:
                nums[l] = nums[r]
                l+=1
            r+=1
        return l